src/main/java/at/hannibal2/skyhanni/features/mining/crystalhollows/CrystalHollowsLobbyAgeWarning.kt

```diff
@@ -0,0 +1,88 @@
+                oldPlayerCount = playerCount //set new cache value
+                if (config.playerCountReminders) LorenzUtils.chat("§a$playerCount players are currently in this Crystal Hollows lobby.")
+                if (!lobbyIsPastAgeThreshold) return
+                if (playerCount in 1..config.minPlayers) LorenzUtils.chat("§cThere are $playerCount players remaining in this Crystal Hollows lobby. §4§lIt will shut down very soon.")
+                else LorenzUtils.chat("There are $playerCount players remaining in this Crystal Hollows lobby. §cIt will shut down when there are fewer than ${config.minPlayers} people left, or when this lobby reaches Day ${config.maxLobbyAgeThreshold} §e(whichever happens first).")
+            }
+        }
+    }
+
+    private fun processLobbyAge() {
+        if (!isInCrystalHollows()) return //avoid incredibly fringe race condition that somehow only happens in the dwarven mines -ery
+        val lobbyAge = getLobbyAgeInMinecraftDays()
+        if (oldLobbyAge == lobbyAge) return //avoid sending chat msgs more than necessary
+        oldLobbyAge = lobbyAge //set new cache value
+        if (config.lobbyAgeReminders && !lobbyIsPastAgeThreshold) LorenzUtils.chat("§aThis Crystal Hollows lobby is currently at Day $lobbyAge, ${abs(config.minLobbyAgeThreshold - lobbyAge)} day(s) away from Day ${config.minLobbyAgeThreshold}.")
+        if (lobbyAge in config.minLobbyAgeThreshold..< config.maxLobbyAgeThreshold && !lobbyIsPastAgeThreshold) {
+            lobbyIsPastAgeThreshold = true //prevent sending message multiple times (see preceding conditional)
+            LorenzUtils.chat("This Crystal Hollows lobby has reached Day ${config.minLobbyAgeThreshold}. It no longer accepts new players, §cand will shut down on Day ${config.maxLobbyAgeThreshold} or when there are fewer than ${config.minPlayers} people left §e(whichever happens first).")
+        } else if (lobbyAge >= config.maxLobbyAgeThreshold) LorenzUtils.chat("§cThis Crystal Hollows lobby has reached Day ${config.maxLobbyAgeThreshold}. §4§lIt will shut down very soon.")
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!isEnabled()) return
+        if (!event.repeatSeconds(1)) return
+        processLobbyAge()
+    }
+
+    @SubscribeEvent
+    fun onTabListUpdate(event: TabListUpdateEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!isEnabled()) return
+        processPlayerCount()
+    }
+
+    @SubscribeEvent
+    fun onIslandChange(event: IslandChangeEvent) {
+        if (event.newIsland != THE_CRYSTAL_HOLLOWS) {
+            lobbyIsPastAgeThreshold = false //allow sending again in next CH lobby
+            oldPlayerCount = 0L //reset cache values
+            oldLobbyAge = 0L //reset cache values
+        }
+    }
+
+    private val config get() = SkyHanniMod.feature.mining.crystalHollowsLobbyAgeWarning
```

Comment:
i feel most of our code has the config variable at the top of the file not the bottom